### PR TITLE
Fixes #5 - Start batch tool with start options

### LINKED ISSUE #5: Start batch tool with start options

### PR DESCRIPTION
**Pull request by rwmcintosh**
Starting the batch tool with the correct options.
Binding to log file output is coming next

### CODE REVIEW

File: src/OSPSuite.InstallationValidator.Core/Services/BatchStarterTask.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
+﻿using System.IO;
+using OSPSuite.InstallationValidator.Core.Assets;
+using OSPSuite.InstallationValidator.Core.Domain;
+using OSPSuite.Utility;
+
+namespace OSPSuite.InstallationValidator.Core.Services
+{
+   public interface IBatchStarterTask
+   {
+      StartableProcess StartValidation(string outputFolderPath);
```
Comment (msevestre) [inline src/OSPSuite.InstallationValidator.Core/Services/BatchStarterTask.cs]: Is this the task that is going to also perform the comparison?
It feels that this should be more of a Runner like we have for the PArameterIdentification and this should return a Task that can be canceled using a CancelationToken.
The actual StartableProcess would not be returned

File: src/OSPSuite.InstallationValidator.Core/Domain/StartableProcess.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      }
+
+      public virtual void Stop()
+      {
+         if (_process == null || _exited)
+            return;
+
+         unsubscribe();
+         _process.Kill();
+         _process.Dispose();
```
Comment (msevestre) [inline src/OSPSuite.InstallationValidator.Core/Domain/StartableProcess.cs]: It is never disposed if you do not call Stop explicitly. At least I did not see how

Comment (msevestre) [inline src/OSPSuite.InstallationValidator.Core/Domain/StartableProcess.cs]: Should not that guy implement IDisposable? THat way it could dispose of the process when the disposed as well

File: src/OSPSuite.InstallationValidator.Core/Presentation/MainPresenter.cs [hunk 3]
```diff
@@ -11,17 +12,20 @@ public interface IMainPresenter : IDisposablePresenter
    {
       void SelectOutputFolder();
       void Abort();
-      Task StartInstallationValidation();
```
Comment (msevestre) [inline src/OSPSuite.InstallationValidator.Core/Presentation/MainPresenter.cs]: Yeah for the Task! Why did you change it?

File: src/OSPSuite.InstallationValidator.Core/Presentation/MainPresenter.cs [hunk 4]
```diff
@@ -35,14 +39,19 @@ public void SelectOutputFolder()
...
 
+      public override void ViewChanged()
+      {
+         _view.OkEnabled = _view.HasError;
+      }
+
       public void Abort()
       {
-         throw new System.NotImplementedException();
+         _batchStarterTask.StopValidation(_process);
```
Comment (msevestre) [inline src/OSPSuite.InstallationValidator.Core/Presentation/MainPresenter.cs]: See above. Use cancelationToken and stop the task etc.. see ParmaeterIdentificationRun in Core I think

File: src/OSPSuite.InstallationValidator.Core/Domain/StartableProcess.cs [hunk 5]
```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Diagnostics;
+
+namespace OSPSuite.InstallationValidator.Core.Domain
+{
+   public class StartableProcess
```
Comment (msevestre) [inline src/OSPSuite.InstallationValidator.Core/Domain/StartableProcess.cs]: This seems to be a very generic class that should probably be moved to Core as well?

File: src/OSPSuite.InstallationValidator.Core/Domain/StartableProcess.cs [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+   {
+      private readonly Process _process;
+      private bool _exited;
+
+      public StartableProcess(string filePath, string arguments = null)
```
Comment (msevestre) [inline src/OSPSuite.InstallationValidator.Core/Domain/StartableProcess.cs]: what about passing a param arrays of arguments and this class is responsible for buiding the argument string? Would that be helpful?

File: src/OSPSuite.InstallationValidator.Core/Presentation/MainPresenter.cs (hunk 3 above)
Comment (rwmcintosh) [inline src/OSPSuite.InstallationValidator.Core/Presentation/MainPresenter.cs]: Reason I changed it was that the started process is already completely asynchronous. There is no need to use the task and cancellation token as I would only use that to call process.kill and process.dispose anyway

File: src/OSPSuite.InstallationValidator.Core/Presentation/MainPresenter.cs (hunk 4 above)
Comment (rwmcintosh) [inline src/OSPSuite.InstallationValidator.Core/Presentation/MainPresenter.cs]: Why do you want to manage that as async? There is no need since the process is completely async anyway no?

**Comment by msevestre** [comment]
Abort is for the whole operation (run, compare etc)
Plus the caller should not know how the process deep down is implemented

**Comment by msevestre** [comment]
The presenter should probably not now much about the process details. There
is the validationrunner (missing at the moment) and that's the only one
that the presenter is talking to.

File: src/OSPSuite.InstallationValidator.Core/Presentation/MainPresenter.cs [hunk 7]
```diff
@@ -35,14 +39,19 @@ public void SelectOutputFolder()
          _outputFolderDTO.FolderPath = outputFolder;
       }
 
+      public override void ViewChanged()
```
Comment (msevestre) [inline src/OSPSuite.InstallationValidator.Core/Presentation/MainPresenter.cs]: This ViewChanged stuff looks weird. 
The view notifies the changes and then a prop on the view is set based on a view state...

File: src/OSPSuite.InstallationValidator.Core/Domain/StartableProcess.cs (hunk 5 above)
Comment (rwmcintosh) [inline src/OSPSuite.InstallationValidator.Core/Domain/StartableProcess.cs]: Added a todo

File: src/OSPSuite.InstallationValidator.Core/Domain/StartableProcess.cs (hunk 6 above)
Comment (rwmcintosh) [inline src/OSPSuite.InstallationValidator.Core/Domain/StartableProcess.cs]: I changed that

File: src/OSPSuite.InstallationValidator.Core/Domain/StartableProcess.cs (hunk 2 above)
Comment (rwmcintosh) [inline src/OSPSuite.InstallationValidator.Core/Domain/StartableProcess.cs]: yeah, you are right. Implementing disposable now

File: src/OSPSuite.InstallationValidator.Core/Presentation/MainPresenter.cs (hunk 7 above)
Comment (msevestre) [inline src/OSPSuite.InstallationValidator.Core/Presentation/MainPresenter.cs]: This actually should be taken care of automatically I believe

File: src/OSPSuite.InstallationValidator.Core/Presentation/MainPresenter.cs (hunk 4 above)
Comment (rwmcintosh) [inline src/OSPSuite.InstallationValidator.Core/Presentation/MainPresenter.cs]: changed to async

File: src/OSPSuite.InstallationValidator.Core/Services/BatchStarterTask.cs (hunk 1 above)
Comment (rwmcintosh) [inline src/OSPSuite.InstallationValidator.Core/Services/BatchStarterTask.cs]: done

File: src/OSPSuite.InstallationValidator.Core/Presentation/MainPresenter.cs (hunk 7 above)
Comment (rwmcintosh) [inline src/OSPSuite.InstallationValidator.Core/Presentation/MainPresenter.cs]: Indeed, although I added that the button is disabled when the batch tool is running, so I changed it around a bit now.

File: src/OSPSuite.InstallationValidator.Core/Domain/StartableProcess.cs [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+      {
+         _process?.Start();
+      }
+
+      public void Dispose()
+      {
+         _process.Dispose();
+      }
+
+      public virtual void Wait(CancellationToken token)
```
Comment (msevestre) [inline src/OSPSuite.InstallationValidator.Core/Domain/StartableProcess.cs]: I have no idea what that all does here. I am assuming you tried it with the cancel and it works?

Comment (rwmcintosh) [inline src/OSPSuite.InstallationValidator.Core/Domain/StartableProcess.cs]: Yes, it works for both normal finish (_exited = true and return) as well as cancel (_process.Kill and throw).

Comment (rwmcintosh) [inline src/OSPSuite.InstallationValidator.Core/Domain/StartableProcess.cs]: I don't follow...
If the user presses cancel, then the process.Kill is called and OperationCanceled is thrown. Not correct?

Comment (rwmcintosh) [inline src/OSPSuite.InstallationValidator.Core/Domain/StartableProcess.cs]: oh, I probably don' t need to throw that. Is that what you mean?

Comment (rwmcintosh) [inline src/OSPSuite.InstallationValidator.Core/Domain/StartableProcess.cs]: gone

Comment (rwmcintosh) [inline src/OSPSuite.InstallationValidator.Core/Domain/StartableProcess.cs]: merging and I'll investigate that for use with process
